src/inspect.mjs

```diff
@@ -1,95 +1,489 @@
 import { surroundingAgent } from './engine.mjs';
 import { Type, Value, wellKnownSymbols } from './value.mjs';
 import {
-  Call, IsArray, Get, LengthOfArrayLike,
+  Assert,
+  Call,
   EscapeRegExpPattern,
+  Get,
+  HasOwnProperty,
+  IsAccessorDescriptor,
+  IsArray,
+  IsCallable,
+  IsDataDescriptor,
+  LengthOfArrayLike,
+  isArrayIndex,
+  isECMAScriptFunctionObject,
 } from './abstract-ops/all.mjs';
 import { Q, X } from './completion.mjs';
+import { OutOfRange } from './helpers.mjs';
 
 const bareKeyRe = /^[a-zA-Z_][a-zA-Z_0-9]*$/;
 
-const getObjectTag = (value, wrap) => {
-  let s;
-  try {
-    s = X(Get(value, wellKnownSymbols.toStringTag)).stringValue();
-  } catch {}
+const defaultOptions = {
+  colors: false,
+  showHidden: false,
+  showProxy: false,
+};
+Object.setPrototypeOf(defaultOptions, null);
+Object.seal(defaultOptions);
+
+const styles = {
+  bigint: 'yellow',
+  boolean: 'yellow',
+  date: 'magenta',
+  hidden: 'blackBright',
+  internalSlot: 'blackBright',
+  module: 'underline',
+  name: undefined,
+  null: 'bold',
+  number: 'yellow',
+  regexp: 'red',
+  special: 'cyan',
+  string: 'green',
+  symbol: 'green',
+  undefined: 'blackBright',
+};
+Object.setPrototypeOf(styles, null);
```

Comment:
you can use `__proto__: null`

Reply:
Well, that’s an **Annex B** feature, so I’d rather wait for <https://github.com/tc39/ecma262/issues/2088#issuecomment-656486631> first.

Reply:
This code doesn't run inside engine262, and `__proto__: null` is used elsewhere.